### ISSUE TITLE
Calamares working on available partitions in mounted disk

### ISSUE THREAD

**Issue by D3vil0p3r** [calamares/calamares#2167]
Currently Calamares is not able to use an available partition inside the mounted disk. For example, if I am on an installed Linux system with 120 GB on `/dev/sda` disk and `/dev/sda1` partition, and by GParted I want to shrink `/dev/sda1` for creating a `/dev/sda2` partition of 60 GB in order to have, initially as unallocated:

<img width="392" alt="image" src="https://github.com/calamares/calamares/assets/83867734/5d52cb7f-6271-4a94-80d9-8dbc88ac7747">

and I run Calamares, it returns:

<img width="373" alt="image" src="https://github.com/calamares/calamares/assets/83867734/eb4eb28c-d989-4a8f-bc5c-994ada84ff5e">

It should occur because Calamares, when running, produces this:
```
14:58:49 [6]: getting smart status failed for  "/dev/sda" :  Success
14:58:49 [6]: error during smart output parsing for  "/dev/sda" :  Success
14:58:49 [6]: smartctl initialization failed for  "/dev/zram0" :  Resource temporarily unavailable
14:58:49 [6]: error during smart output parsing for  "/dev/zram0" :  Success
14:58:49 [6]: QList<Device*> PartUtils::getDevices(DeviceType)
    Removing unsuitable devices: 2 candidates. 
    .. Removing device with root filesystem (/) on it "/dev/sda" 
    .. Removing zram "/dev/zram0" 
    .. there are 0 devices left. 
14:58:49 [6]: void PartitionCoreModule::doInit()
    LIST OF DETECTED DEVICES: 
    .. node	capacity	name	prettyName 
    .. 0 devices detected. 
```
so it seems to remove the mounted disk.

It means that I can use Calamares ONLY if I have it on an ISO burned in an external media drive, and I cannot use Calamares if I'm working on an already installed Linux system.

Is it possible to make Calamares working also when I am on the same disk where I created a new partition for installing the new OS?

**Comment by dalto8** [comment]
Yeah, this is a well known issue that has been reported many times.  For example, here is one from the other day: https://github.com/calamares/calamares/issues/2157

There are already open issues for it so I am going to close this as a **duplicate**.

**Comment by D3vil0p3r** [comment]
@dalto8 thank you for the answer. I see that the first issue ticket is on 2016. On all this time is there no way to implement/fix it?
